example/client.rb

```diff
@@ -397,26 +414,38 @@ def webhook_parse_example
       }
     JSON
 
+    webhook_content = ResponseStruct.new(
+      headers,
+      body,
+      "<url>"
+    )
+
     webhook_api = Bitpesa::ApiClient.new
-    webhook = webhook_api.parse_response(webhook_content, "Webhook")
-
-    if webhook['event'].start_with?('transaction')
-      transaction_webhook = webhook_api.parse_response(webhook_content, 'TransactionWebhook')
-      puts transaction_webhook.object.to_s
-    elsif webhook['event'].start_with?('recipient')
-      recipient_webhook = webhook_api.parse_response(webhook_content, 'RecipientWebhook')
-      puts recipient_webhook.object.to_s
-    elsif webhook['event'].start_with?('payout_method')
-      payout_method_webhook = webhook_api.parse_response(webhook_content, 'PayoutMethodWebhook')
-      puts payout_method_webhook.object.to_s
-    elsif webhook['event'].start_with?('sender')
-      sender_webhook = webhook_api.parse_response(webhook_content, 'SenderWebhook')
-      puts sender_webhook.object.to_s
-    elsif webhook['event'].start_with?('document')
-      document_webhook = webhook_api.parse_response(webhook_content, 'DocumentWebhook')
-      puts document_webhook.object.to_s
-    else
-      puts webhook
+    webhook = webhook_api.parse_response(webhook_content.body, "Webhook")
+    nonce = JSON.parse(webhook_content.headers)['Authorization-Nonce']
```

Comment:
as above: no need for this complex setup. It's enough to have nonce = headers['Authorization-Nonce'], etc. We can expect that the headers will be in a nice format here as most client libraries will return them in a hash, and never in JSON